### PR TITLE
[5.5] Remove usages of the with() helper

### PR DESCRIPTION
**Pull request by KennedyTedesco**
Now that Laravel 5.5 requires PHP 7 the usage of `with()` becomes unnecessary.

### CODE REVIEW

File: src/Illuminate/Support/helpers.php [hunk 1]
```diff
@@ -924,16 +924,3 @@ function windows_os()
         return strtolower(substr(PHP_OS, 0, 3)) === 'win';
     }
 }
-
-if (! function_exists('with')) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: We shouldn't delete this helper yet.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/helpers.php (hunk 1 above)
Comment (KennedyTedesco) [inline src/Illuminate/Support/helpers.php]: This helper was built to support some internal needs of the core, wasn't even documented for the users. And, after all, is for the next release. Looks reasonable to me to move forward.

I'll wait for more feedbacks about this. Thanks.

**Comment by taylorotwell** [comment]
Yeah, let's leave the helper in place. The rest looks fine.

**Comment by KennedyTedesco** [comment]
@taylorotwell As requested, done.

**Review by GrahamCampbell — APPROVED** [review]

**Pull request merged**